### ISSUE TITLE
javadoc and sources jar task names should be configurable

### ISSUE THREAD

**Issue by rpalcolea** [gradle/gradle#12215]
Hi folks,

Gradle 6 introduced Built-in javadoc and sources packaging and publishing via

```
java {
    withJavadocJar()
    withSourcesJar()
}
```

While this is fantastic, we have found ourselves in the need for changing the default task name for one of them.

Historically we leveraged our own sources jar task under `sourceJar` which is not populated across several projects and there are specific configurations like `archiveBaseName`

it would be phenomenal if we could have a way to set the task name for these new tasks.

Something like the following could be great for our use case:

```
java {
  withJavadocJar('myJavaDocJar')
  withSourcesJar('sourceJar')
}
```

This is a naive search for usage of `sourceJar` -> https://github.com/search?l=Gradle&q=sourceJar&type=Code

**Comment by ljacomet** [comment]
The whole reason we made these two features an opt-in was that we knew existing builds had to do it on their own, with potentially colliding task names.

So the idea is that you opt-in when you are ready to replace the custom tasks with those provided by these entry points. Note that you can still configure the tasks, they are registered immediately.

So I fail to see what the name change would give you. Keep two different ways of producing a source jar in a single project?

**Comment by rpalcolea** [comment]
Hi @ljacomet ,

So the current use case for us is the following: https://github.com/nebula-plugins/nebula-publishing-plugin/blob/master/src/main/groovy/nebula/plugin/publishing/publications/SourceJarPlugin.groovy#L37

We created a `sourceJar` task years ago and we have found several projects where folks configure the task to modify things like the `archiveBaseName`, ex.

```
sourceJar.baseName = 'my-shadow-artifact'
```

I believe our goal is to stay the closest to Gradle OSS offerings so removing custom code for a source jar task and use the new Gradle features is something we would love to do.

With that said, we could try to migrate people to refactor their build files although it isn't a nice experience for them

That's why I was thinking that providing a way to set the task name would be great as it would ease migration for many other folks. Don't know if that makes sense.

Unfortunately, we can't go and include the new one and keep the old one because we would have colliding artifacts for the same module/classifier.

At some point, we would have to see also how shadow plugin adds its own variants for the sources
